### PR TITLE
Add join challenge modal

### PR DESCRIPTION
**Pull request by sebastiantoh**
image

Feel free to let me know if there's any styling changes you want me to make!

### CODE REVIEW

**Comment by jialin7878** [comment]
thanks so much!!!

> Feel free to let me know if there's any styling changes you want me to make!

maybe can add a note in the modal to notify User that they won't be able to change their recurring days upon submission?
thinking if we need to explain to them what recurring days mean as well but it might clutter the modal, maybe a question mark icon button beside recurring days(?)

summoning our great @ming-00 what do u think

File: frontend/src/pages/challenge/ChallengeDetailsPage.tsx [hunk 1]
```diff
@@ -183,7 +186,22 @@ const ChallengeDetailsPage: React.FC = () => {
...
+        {!userChallenge && (
+          <>
+            <Button onClick={() => setIsScheduleModalOpen(true)}>
+              Join Challenge!
+            </Button>
+            <ScheduleModal
+              isOpen={isScheduleModalOpen}
+              onClose={() => setIsScheduleModalOpen(false)}
+              onSubmit={(schedule: Schedule) =>
+                dispatch(joinChallenge(Number(challengeId), schedule))
```
Comment (jialin7878) [inline frontend/src/pages/challenge/ChallengeDetailsPage.tsx:200]: do we need a check here for our three challenges limit?

**Comment by sebastiantoh** [comment]
> thanks so much!!!
> 
> > Feel free to let me know if there's any styling changes you want me to make!
> 
> maybe can add a note in the modal to notify User that they won't be able to change their recurring days upon submission? thinking if we need to explain to them what recurring days mean as well but it might clutter the modal, maybe a question mark icon button beside recurring days(?)
> 
> summoning our great @ming-00 what do u think

image

image

Like this? Feel free to change the copywriting / phrasing!

File: frontend/src/pages/challenge/ChallengeDetailsPage.tsx (hunk 1 above)
Comment (sebastiantoh) [reply frontend/src/pages/challenge/ChallengeDetailsPage.tsx:201]: yep! i'll add this as a TODO to be resolved next time. I don't think we fetch all challenges (or the number of challenges) the user is enrolled in atm

File: backend/db/seeds.rb [hunk 2]
```diff
@@ -54,3 +59,9 @@
                   completed_at: i < 2.weeks.in_days ? 2.weeks.ago + i.days : nil,
                   scheduled_for: 2.weeks.ago + i.days)
 end
+
+30.times do |i|
+  description = "Write #{(i + 1).ordinalize} pages today"
+  Task.create(challenge_id: challenge_running.id, name: "Day #{i + 1}: What's up",
```
Comment (sebastiantoh) [inline backend/db/seeds.rb:65]:
```suggestion
  Task.create(challenge_id: challenge_journalling.id, name: "Day #{i + 1}: What's up",
```

**Pull request merged**